### PR TITLE
Add wrapper class for inception model and file utility.

### PR DESCRIPTION
**Pull request by MarkAZhang**
Adds the following files:
* An InceptionMosquitoModel that creates, trains, and exports an Inception model.
* File utilities that take a directory of images (with subdirectories corresponding to different categories) and splits them up into directories for training, validation, and test based on defined ratios.
* A notebook with code for visualization predictions of the trained model, including all predictions the model made incorrectly.

### CODE REVIEW

File: cli/inception_mosquito_model.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        self.val_image_dir = val_image_dir
+        self.fetch_image_data()
+        self.initialize_model()
+        self.start_time = time.time()
+
+    def fetch_image_data(self):
+        # Includes data augmentation parameters.
+        # The model actually performs better on the validation data without these parameters,
+        # so disable them if needed.
+        image_generator_train = tf.keras.preprocessing.image.ImageDataGenerator(
```
Comment (pranathivemuri) [inline cli/inception_mosquito_model.py]: TODO: move to config file

File: cli/inception_mosquito_model.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+from tensorflow.keras import layers
+import logging
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger("inception_mosquito_model")
+
+IMAGE_DIM = 224
+TENSORFLOW_FEATURE_URL = "https://tfhub.dev/google/imagenet/inception_v3/feature_vector/4"
+EPOCHS = 200
```
Comment (pranathivemuri) [inline cli/inception_mosquito_model.py]: TODO: move to config file

File: cli/inception_mosquito_model.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            input_shape = [IMAGE_DIM, IMAGE_DIM, 3],
+            trainable = False
+        )
+        self.model = tf.keras.Sequential([
+            feature_extractor_layer,
+            layers.Dense(self.train_image_data.num_classes, activation='softmax')
+        ])
+
+        self.model.compile(
+            optimizer=tf.keras.optimizers.Adam(),
```
Comment (pranathivemuri) [inline cli/inception_mosquito_model.py]: You can also change these 3 parameters and see how your model performs. 

adagrad optimizer is supposed to be better I think?

anyway I think these 3 params here can be moved to config file as well

https://github.com/czbiohub/microDL/blob/master/micro_dl/config_lr_finder.yml - example config file

File: cli/inception_mosquito_model.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+class InceptionMosquitoModel:
+    def __init__(self, train_image_dir, val_image_dir):
+        self.train_image_dir = train_image_dir
+        self.val_image_dir = val_image_dir
+        self.fetch_image_data()
+        self.initialize_model()
+        self.start_time = time.time()
+
+    def fetch_image_data(self):
+        # Includes data augmentation parameters.
```
Comment (pranathivemuri) [inline cli/inception_mosquito_model.py]: increase epochs for data augmented model and look at the loss curve again

**Comment by pranathivemuri** [comment]
great looking code!

**Comment by pranathivemuri** [comment]
@MarkAZhang when you are free, if you format this code to accept a config file where you can tune all the parameters for data augmentation etc, then we can run different models and compare. 

if you want to merge this basic code into master and do the config files in the next PR that's cool too.

here is a config file example that has data augmentation parameters - https://github.com/czbiohub/luminoth-uv-imaging/blob/master/luminoth/models/ssd/base_config.yml#L83

**Comment by MarkAZhang** [comment]
Will add the config file in a follow-up PR!
